packages/examples/advanced-react/src/hooks/useWallet.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  SupportedChains,
+  useWallet,
+} from "@civic/multichain-connect-react-core";
+import { WalletContextState } from "@solana/wallet-adapter-react";
+import { Wallet as EthersWallet } from "ethers";
+import { useState, useEffect } from "react";
+
+export interface MultiWalletContextState {
+  connected: boolean;
+  solanaWallet?: WalletContextState;
+  ethersWallet?: EthersWallet;
+  address?: string;
+}
+
+export const useMultiWallet = (): MultiWalletContextState => {
+  const [address, setAddress] = useState<string | undefined>(undefined);
+  const { connected } = useWallet();
+
+  const { wallet: solanaWallet } = useWallet<
+    SupportedChains.Solana,
+    WalletContextState,
+    never
+  >();
+
+  const { wallet: ethersWallet } = useWallet<
```

Comment:
I find having to do this as a user is quite clunky and non-intuitive, especially the 'never' casting. I don't see how this is better than calling a 'useWallet' function from rainbowkit directly.

Reply:
The whole purpose of the lib is so the user can have a single hook to retrieve the connected wallet. If the user needs to use multiple hooks that are dependant on the wallet the user has connected with, I don't see much of a benefit to using our library vs using the adapters we have wrapped internally. 

The idea is to have a single interface and when the user chooses to interact with the specific adapter implementation, they can cast to it. 

I don't see a problem in exposing those hooks if you think it would be beneficial?